cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/OpenAI.scala

```diff
@@ -299,14 +339,46 @@ class OpenAICompletion(override val uid: String) extends CognitiveServicesBase(u
         getValueOpt(r, logProbs).map(v => ("logprobs", v))
       ).flatten).toMap
 
-      getValueOpt(r, prompt).map { prompt =>
-        val fullPayload = optionalParams.updated("prompt", prompt)
-        new StringEntity(fullPayload.toJson.compactPrint, ContentType.APPLICATION_JSON)
-      }
+      getValueOpt(r, prompt)
+        .map(prompt => getStringEntity(prompt, optionalParams))
+        .orElse(getValueOpt(r, bulkPrompt)
+          .map(bulkPrompt => getArrayStringEntity(bulkPrompt, optionalParams)))
+        .orElse(getValueOpt(r, indexPrompt)
+          .map(indexPrompt => getIndexStringEntity(indexPrompt, optionalParams)))
+        .orElse(getValueOpt(r, bulkIndexPrompt)
+          .map(bulkIndexPrompt => getIndexArrayStringEntity(bulkIndexPrompt, optionalParams)))
+        .orElse(throw new IllegalArgumentException(
+          "Payload needs to contain String, Seq[String], Seq[Int] or Seq[Seq[Int]]. This code should not run"))
```

Comment:
>Payload needs to contain String, Seq[String], Seq[Int] or Seq[Seq[Int]]. [](http://example.com/codeflow?start=11&length=72)

This probably needs better wording. Suggestions?

Reply:
The prompt field should be set with a String, Seq[String], Seq[Int] or Seq[Seq[Int]], received ${.....}.